### ISSUE TITLE
Rozdělit projekt na moduly

### ISSUE THREAD

**Issue by encukou** [PyLadiesCZ/roboprojekt#21]
Přidat moduly:
* [x] Backend
* [x] Frontend
* [x] Testy
* [x] Hra

Až to bude hotové, smazat skript kde je to teď všechno.

**Comment by befeleme** [comment]
Čaute, návrh níže. Otázníky jsou tam, kde si nejsem vůbec jistá tím řešením a uvítám komentář. Uvítám komentář i tam, kde otázníky nejsou. Názvy funkcí jsou pouze pracovní.

- moduly volají backend, ale backend nevolá žádný herní modul (vyhýbame se cyklickým importům, což je OK)
- bavili jsme se, že game zavola logiku, vykreslení a vykreslí vše - mne zatím připadalo, že hře stačí zavolat frontend a udělá, co má, takže nejspíš jsem na něco pozapomněla
- k modulu backend patří zpracování rotace a vrstev, frontend si musí poradit s jejich vykreslením.
- umím si trochu představit testování backendu, vůbec neumím frontendu

Kdo si vezmete na startost backend, frontend a (testy, hru)? Myslím si, že se může na tom pracovat do jisté míry souběžně (frontend může zatím načítat zkušební set dat), hra a testy až na konci.

BACKEND: backend.py (je to validní název souboru?)
```
import json
read_json_map(map_name)
    returns: data
    
create_coordinates(map_width, map_height) # pocet policek
    returns: list of tuples (x, y)
    
find_tiles_path(data, tj. tilesets)  # sahnout do jsonu
    returns: dictionary tile_id -> image path
    
create_board(list of tuples (x, y), list of tile effects) # tile effects = nas seznam z jsonu
    returns dictionary tuple -> tile effect 
```  

FRONTEND  frontend.py
```
import pyglet, backend

funkce:
draw_board(board_dictionary, tiles_path, window size) 
    returns: list of loaded sprites 
    
pygleti on_draw??? patri to sem???
```
  
GAME game.py
```
import frontend, pyglet

pyglet.app.run() ???
```

TESTS tests.py
`import pytest, backend`

**Comment by anezkamll** [comment]
Já si klidně zkusím vzít backend, můžu na tom pracovat zítra, v pátek a trochu v sobotu. 

Chápu to správně, že jako výchozí nyní bereme kód od @befeleme, který se bude postupně vylepšovat?

**Comment by befeleme** [comment]
> Chápu to správně, že jako výchozí nyní bereme kód od @befeleme, který se bude postupně vylepšovat?

Jo, ten by se měl postupně přepsat, a původní soubor se na konci smaže.

**Comment by encukou** [comment]
> bavili jsme se, že game zavola logiku, vykreslení a vykreslí vše - mne zatím připadalo, že hře stačí zavolat frontend a udělá, co má, takže nejspíš jsem na něco pozapomněla

Frontend by neměl umět načítat data (mapy ze souboru). Měl by je dostat už načtené. Hra tedy musí na to načtení zavolat backend.

> umím si trochu představit testování backendu, vůbec neumím frontendu

Na frontend a hru (zatím) testy nepište.

> GAME game.py

```python
import backend, frontend, pyglet

board = backend.read_json_map(...)

… pygletí on_draw (které jen volá frontend)

pyglet.app.run()
```

Na testy je zvláštní issue; nechte je na sobotu/neděli.

**Comment by befeleme** [comment]
Ok, holky @MyshiczkaH, @hranka - ktera si vezmete na starost frontend?

**Comment by MyshiczkaH** [comment]
@befeleme  zkusím, mám doma poradce tak mě kdyžtak nakopne :-)

**Comment by anezkamll** [comment]
Dotaz - co preferujete? Všechny pracovní verze mám dávat průběžně sem, nebo sem mám hodit až nějakou, co bude vypadat co k čemu? 
Příklad: Teď si zkopíruju Karolinin kód, odmažu z toho věci, co backend neřeší, udělám si do toho nějaké pracovní komentáře a pak budu muset práci přerušit a pokračovat třeba o den později. Chcete, abych sem hodila i tohle, nebo až nějakou smysluplnou rozpracovanou verzi?

**Comment by encukou** [comment]
K rozdělaným pull requestům se dává zkratka "WIP" (Work in Progress). Někdy jako [štítek](https://github.com/python/cpython/pull/5141), někdy se to dá do názvu ([příklad](https://github.com/python/cpython/pull/5141)). Ostatní pak můžou komentovat směr, kam se ubíráš, ale ví že to není hotové.

**Comment by encukou** [comment]
Může! Zvlášť když je v něm napsáno že není hotový.
Bez začlenění by bylo těžké ty funkce volat odjinud :)

**Comment by befeleme** [comment]
@MyshiczkaH, diky za poslani frontendu + hry.

**Comment by befeleme** [comment]
Tests.py pribudou spolu s prvnimi testy od Zuzky.

**Comment by befeleme** [comment]
Vše začleněno, děkuji.

**Comment by encukou** [comment]
Pořád tu vidím původní skript.
